File: src/Infraestructure/Class/PisoGarage.ts
import { IEspacio } from "../Interfaces/IEspacio";
import { IPisoGarage } from "../Interfaces/IPisoGarage";
import { Espacio } from "./Espacio";
import { NoVehiculo } from "./Vehiculos/NoVehiculo";

export class PisoGarage implements IPisoGarage{
    public CantidadEspacios: number;
    public Espacios: Array<IEspacio>;

    public constructor(cantidadEspacios: number, piso: number){
        this.CantidadEspacios = cantidadEspacios;
        this.Espacios = new Array<IEspacio>(cantidadEspacios);

        for (var i = 0; i < this.Espacios.length; i++) {
            this.Espacios[i] = new Espacio(piso, i, false, new NoVehiculo());
        }
    }
}

File: src/main/java/Grupo2/Garage/Infraestructure/Class/Garage/PisoGarage.java
package Grupo2.Garage.Infraestructure.Class.Garage;

import Grupo2.Garage.Infraestructure.Class.Vehiculos.NoVehiculo;
import lombok.var;

import java.util.ArrayList;

public class PisoGarage {
    public Integer CantidadEspacios;
    public ArrayList<Espacio> Espacios;

    public PisoGarage(Integer cantidadEspacios, Integer piso){
        this.CantidadEspacios = cantidadEspacios;
        this.Espacios = new ArrayList<Espacio>(cantidadEspacios);

        for (var i = 0; i < cantidadEspacios; i++) {
            this.Espacios.add(i, new Espacio(piso, i, false, new NoVehiculo()));
        }
    }
}
